spec/javascripts/components/cookie-banner-new-spec.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-env jasmine, jquery */
+/* global GOVUK */
+
+var container, element;
+
+var GOVUK = window.GOVUK || {};
+
+describe('New cookie banner', function () {
+  'use strict'
+
+  beforeEach(function () {
+    container = document.createElement('div')
+    container.innerHTML =
+      '<div id="global-cookie-message" class="gem-c-cookie-banner--new" data-module="cookie-banner">' +
+        '<div class="gem-c-cookie-banner__wrapper govuk-width-container" data-cookie-banner-main="true">' +
+          '<p class="gem-c-cookie-banner__message">GOV.UK uses cookies to make the site simpler.</p>' +
+          '<div class="gem-c-cookie-banner__buttons">' +
+            '<button class="gem-c-button govuk-button gem-c-button--secondary-quiet gem-c-button--inline" type="submit" data-module="track-click" data-accept-cookies="true" data-track-category="cookieBanner" data-track-action="Cookie banner accepted">Accept cookies</button>' +
+            '<a class="gem-c-button govuk-button gem-c-button--secondary-quiet gem-c-button--inline" role="button" data-module="track-click" data-track-category="cookieBanner" data-track-action="Cookie banner settings clicked" href="/help/cookies">Cookie settings</a>' +
+          '</div>' +
+        '</div>' +
+        '<div class="gem-c-cookie-banner__confirmation govuk-width-container" data-cookie-banner-confirmation="true" style="display: none;">' +
+          '<p class="gem-c-cookie-banner__confirmation-message">' +
+            'You have accepted all cookies' +
+          '</p>' +
+          '<button class="gem-c-cookie-banner__hide-button" data-hide-cookie-banner="true">Hide</button>' +
+        '</div>' +
+      '</div>'
+
+    document.body.appendChild(container)
+    element = document.querySelector('.gem-c-cookie-banner--new')
+
+    // For some reason, JSON.parse on the cookie works in the browser, but fails in Jasmine tests.
+    // It seems to be due to extra escaping of quotes when the code is run in the tests, which means JSON.parse doesn't
+    // work as expected. So we'll stub this value instead.
+    spyOn(JSON, "parse").and.returnValue({"essential":true,"settings":true,"usage":true,"campaigns":true});
```

Comment:
Is this passing the linter? Appears to be missing some spacing in the JSON.

Reply:
I don't think the spec folder gets linted.

Reply:
https://github.com/alphagov/govuk_publishing_components/blob/master/package.json#L6